// tm/ddAPI.js
const https = require('https');
const StringDecoder = require('string_decoder').StringDecoder;
const decoder = new StringDecoder('utf8');
const lbtargetpath = '/TM/cp/cpgetleavebalance.action?searchTable=leavebalance';
const vacationAddingpath = '/TM/cp/cpgetleavebalance.action?searchTable=leavebalance';
var access_token_options = {
	hostname : 'oapi.dingtalk.com',
	port : 443,
	path : '/gettoken?corpid=ding28029cf6368a0723&corpsecret=f_w7lWlR391E08xkuZh7DoLHuud4YKHzj9cmJrP4DIosgrM55Lwh-Rdv1rJ9jLxe',
	method : 'GET'
};
var jsapi_ticket_options = {
	hostname : 'oapi.dingtalk.com',
	port : 443,
	path : '/get_jsapi_ticket?access_token=',
	method : 'GET'
}
var access_token;
exports.getAccessToken = function (params, callback) {
	var a = https.request(access_token_options, function (res) {
			res.on('data', function (d) {
				result = decoder.write(d)
					access_token = JSON.parse(result).access_token
					console.log('access_token: ', access_token);
				callback(result);
			});
		})
		a.end();
}
exports.getjsapi_ticket = function (params, callback) {
	if (access_token === undefined) {
		console.log('acctoken undefined'+access_token)
		this.getAccessToken('', function () {
			requestjsapi_ticket(callback);
		});
	} else {
		console.log('acctoken defined'+access_token)
		requestjsapi_ticket(callback);
	}

}
requestjsapi_ticket = function (callback) {
	var path = jsapi_ticket_options.path;
	console.log('access_token: ', access_token);
	jsapi_ticket_options.path = path + access_token;
	var a = https.request(jsapi_ticket_options, function (res) {
			res.on('data', function (d) {
				result = decoder.write(d)
					console.log('jsapi_ticket: ', result);
				callback(result);
			});
		})
		jsapi_ticket_options.path = path;
	a.end();
}
exports.access_token;
